packages/webpack-extraction-plugin/src/sortCSSRules.ts

```diff
@@ -1,99 +1,54 @@
-  if (element.type === RULESET || element.type === KEYFRAMES) {
-    return element.value + suffix;
-  }
-
-  if (Array.isArray(element.children)) {
-    return element.value + '[' + element.children.map(child => getElementReference(child, suffix)).join(',') + ']';
-  }
-
-  function removeRootProperty(element: Element): unknown {
-    return {
-      ...element,
-      children: Array.isArray(element.children)
-        ? element.children.map(child => removeRootProperty(child))
-        : element.children,
-      root: undefined,
-      parent: undefined,
-    };
-  }
-
-  throw new Error(
-    [
-      'getElementReference(): An unhandled case, please report if it happens and provide debug information about an element:',
-      JSON.stringify(removeRootProperty(element), null, 2),
-    ].join('\n'),
-  );
-}
-
-export function getStyleBucketNameFromElement(element: Element): StyleBucketName {
-  if (element.type === KEYFRAMES) {
-    return 'k';
-  }
-
-  if (isResetClassName(element.value)) {
-    return 'r';
-  }
-
-  return getStyleBucketName(
-    [getSelectorFromElement(element)],
-    element.type === '@layer' ? element.value : '',
-    element.type === MEDIA ? element.value : '',
-    element.type === SUPPORTS ? element.value : '',
-  );
-}
-
-export function sortCSSRules(css: string, compareMediaQueries: GriffelRenderer['compareMediaQueries']): string {
-  const childElements = compile(css)
-    // Remove top level comments as it is unclear how to sort them
-    .filter(element => element.type !== COMMENT)
-    .map(element => ({
-      ...element,
-      bucketName: getStyleBucketNameFromElement(element),
-      metadata: getElementMetadata(element),
-      reference: getElementReference(element),
-    }));
-  const uniqueElements = childElements.reduce<Record<string, typeof childElements[0]>>((acc, element) => {
-    acc[element.reference] = element;
-
-    return acc;
-  }, {});
-  const sortedElements = Object.values(uniqueElements).sort((elementA, elementB) => {
-    if (elementA.bucketName === 'm' && elementB.bucketName === 'm') {
-      return compareMediaQueries(elementA.metadata, elementB.metadata);
-    }
-
-    if (elementA.bucketName === elementB.bucketName) {
-      return 0;
-    }
-
-    return styleBucketOrdering.indexOf(elementA.bucketName) - styleBucketOrdering.indexOf(elementB.bucketName);
-  });
-
-  return serialize(sortedElements, stringify);
+export function sortCSSRules(
+  setOfCSSRules: CSSRulesByBucket[],
+  compareMediaQueries: GriffelRenderer['compareMediaQueries'],
+) {
+  return styleBucketOrdering
+    .map(styleBucketName => {
+      return {
+        styleBucketName,
+        cssBucketEntries: Object.values(
+          Object.fromEntries(
+            setOfCSSRules.flatMap(cssRulesByBucket => {
+              if (Array.isArray(cssRulesByBucket[styleBucketName])) {
+                return cssRulesByBucket[styleBucketName]!.map(bucketEntry => [
+                  getCSSRuleFromBucketEntry(bucketEntry),
```

Comment:
what's the point of the return of `getCSSRuleFromBucketEntry` here? You simply discard it after calling `Object.values`

Reply:
Makes sense, could you please add some documentation? It's really not obvious what the intention is there

Reply:
Add a comment